libs/shared/typescript/charts/src/lib/boxplot-chart/boxplot-chart.ts

```diff
@@ -0,0 +1,299 @@
+import {
+  DatasetComponentOption,
+  ECharts,
+  EChartsOption,
+  SeriesOption,
+} from 'echarts';
+import { BoxplotProps, CategoryPoint } from '../models';
+import {
+  addXAxisValueToBoxplotSummaries,
+  addXAxisValueToCategoryPoint,
+  formatCategoryPointsForBoxplotTransform,
+  getCategoryPointColor,
+  getCategoryPointShape,
+  getUniqueValues,
+  initChart,
+  setNoDataOption,
+} from '../utils';
+import { XAxisLabelTooltips } from '../x-axis-label-tooltips';
+
+const titleTextStyle = {
+  fontWeight: 700,
+  fontSize: '18px',
+  color: 'black',
+};
+
+const boxplotStyle = {
+  borderColor: '#bcc0ca',
+  borderWidth: 2,
+};
+
+const yAxisPadding = 0.2;
+const defaultPointShape = 'circle';
+const defaultPointSize = 18;
+const defaultPointColor = '#8b8ad1';
+
+export class BoxplotChart {
+  chart: ECharts | undefined;
+  xAxisLabelTooltips: XAxisLabelTooltips | undefined;
+
+  constructor(
+    chartDom: HTMLDivElement | HTMLCanvasElement,
+    boxplotProps: BoxplotProps
+  ) {
+    this.chart = initChart(chartDom);
+    this.setOptions(boxplotProps);
+  }
+
+  destroy() {
+    this.chart?.dispose();
+  }
+
+  setXAxisLabelTooltips(xAxisCategoryToTooltipText?: Record<string, string>) {
+    if (!xAxisCategoryToTooltipText || !this.chart) return;
+    if (!this.xAxisLabelTooltips) {
+      this.xAxisLabelTooltips = new XAxisLabelTooltips(
+        this.chart,
+        xAxisCategoryToTooltipText
+      );
+    }
+    this.xAxisLabelTooltips.setXAxisCategoryToTooltipText(
+      xAxisCategoryToTooltipText
+    );
+  }
+
+  setOptions(boxplotProps: BoxplotProps) {
+    if (!this.chart) return;
+
+    const {
+      points,
+      summaries,
+      title,
+      xAxisTitle,
+      yAxisTitle,
+      yAxisMin,
+      yAxisMax,
+      xAxisCategoryToTooltipText,
+      pointTooltipFormatter,
+    } = boxplotProps;
+
+    if (points.length === 0) {
+      setNoDataOption(this.chart);
+      return;
+    }
+
+    const xAxisCategories = getUniqueValues(
+      points,
+      'xAxisCategory'
+    ) as string[];
+    const pointCategories = getUniqueValues(
+      points,
+      'pointCategory'
+    ) as string[];
+    const hasPointCategories = pointCategories.length > 0;
+
+    const dataForScatterPoints = addXAxisValueToCategoryPoint(
+      points,
+      xAxisCategories,
+      pointCategories,
+      0.05
+    );
+    const dataForStaticBoxplotSummaries = summaries
+      ? addXAxisValueToBoxplotSummaries(summaries, xAxisCategories)
+      : undefined;
+    const dataForDynamicBoxplotTransforms =
+      formatCategoryPointsForBoxplotTransform(
+        dataForScatterPoints,
+        xAxisCategories
+      );
```

Comment:
ECharts can calculate the boxplot summaries based on the provided points data, but the data needs to be formatted as a 2d array where each row contains all the values for a particular category, rather than an array of objects.